File: LeetCode/223-rectangle-area/ex_223.cpp
class Solution {
public:
    int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int interaction = 0;
        int r1 = (C - A) * (D - B);
        int r2 = (G - E) * (H - F);
        
        if (C > E && G > A && D > F && H > B)
            interaction = (min(C, G) - max(A, E)) * (min(D, H) - max(B, F));
        
        return r1 + r2 - interaction;
    }
};

File: LeetCode/223-rectangle-area/ex_223.py
class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """
        rect1 = (C - A) * (D - B)
        rect2 = (G - E) * (H - F)

        if C > E and D > F and G > A and H > B:
            intersection = (min(C, G) - max(E, A)) * (min(D, H) - max(F, B))
        else:
            intersection = 0
        return rect1 + rect2 - intersection
